fdbsyncd/fdbsync.cpp

```diff
@@ -0,0 +1,603 @@
+        return false;
+    }
+    imet_route[key].vni =  vni;
+    return true;
+}
+
+bool FdbSync::checkDelImet(string key, uint32_t vni)
+{
+    int ret = true;
+
+    SWSS_LOG_DEBUG("Del IMET key:%s Vni:%d", key.c_str(), vni);
+    if (imet_route.find(key) != imet_route.end())
+    {
+        SWSS_LOG_DEBUG("IMET found key:%s Vni:%d", key.c_str(), vni);
+        if (imet_route[key].redist_add_status == false)
+        {
+            SWSS_LOG_DEBUG("IMET found key:%s Vni:%d, it should have not happened", key.c_str(), vni);
+            ret = false;
+        }
+        imet_route.erase(key);
+    }
+    return ret;
+}
+
+void FdbSync::imetAddRoute(struct in_addr vtep, string vlan_str, uint32_t vni)
+{
+    string vlan_id = "Vlan" + vlan_str;
+    string key = vlan_id + ":" + inet_ntoa(vtep);
+
+    if (!checkImetExit(key, vni))
+        return;
+
+	SWSS_LOG_NOTICE("%sIMET Add route key:%s vtep:%s %s", 
+			m_AppRestartAssist->isWarmStartInProgress() ? "WARM-RESTART:" : "",
+			key.c_str(), inet_ntoa(vtep), vlan_id.c_str());
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple f("vni", to_string(vni));
+    fvVector.push_back(f);
+
+    imet_route[key].redist_add_status = true;
+    
+	// If warmstart is in progress, we take all netlink changes into the cache map
+    if (m_AppRestartAssist->isWarmStartInProgress())
+    {
+        m_AppRestartAssist->insertToMap(APP_VXLAN_REMOTE_VNI_TABLE_NAME, key, fvVector, false);
+        return;
+    }
+    
+	m_imetTable.set(key, fvVector);
+    return;
+}
+
+void FdbSync::imetDelRoute(struct in_addr vtep, string vlan_str, uint32_t vni)
+{
+    string vlan_id = "Vlan" + vlan_str;
+    string key = vlan_id + ":" + inet_ntoa(vtep);
+
+    if (!checkDelImet(key, vni))
+        return;
+
+	SWSS_LOG_NOTICE("%sIMET Del route key:%s vtep:%s %s", 
+			m_AppRestartAssist->isWarmStartInProgress() ? "WARM-RESTART:" : "", 
+			key.c_str(), inet_ntoa(vtep), vlan_id.c_str());
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple f("vni", to_string(vni));
+    fvVector.push_back(f);
+
+    // If warmstart is in progress, we take all netlink changes into the cache map
+    if (m_AppRestartAssist->isWarmStartInProgress())
+    {
+        m_AppRestartAssist->insertToMap(APP_VXLAN_REMOTE_VNI_TABLE_NAME, key, fvVector, true);
+        return;
+    }
+    
+	m_imetTable.del(key);
+    return;
+}
+
+void FdbSync::macUpdateVxlanDB(string key)
+{
+    string vtep = m_mac[key].vtep;
+    string type;
+    string vni = to_string(m_mac[key].vni);
+
+    if (m_mac[key].type)
+        type = "static";
+    else
+        type = "dynamic";
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple rv("remote_vtep", vtep);
+    FieldValueTuple t("type", type);
+    FieldValueTuple v("vni", vni);
+    fvVector.push_back(rv);
+    fvVector.push_back(t);
+    fvVector.push_back(v);
+
+	SWSS_LOG_NOTICE("%sVXLAN_FDB_TABLE: ADD_KEY %s vtep:%s type:%s", 

```

Comment:
Change to INFO for all frequent events that happens like FDB learning etc. There is ratelimiting for syslogs and we will start losing relevent messages